### PR TITLE
fix for preserve cookie on test suite execution

### PR DESCRIPTION
**Pull request by mauceballos**
|Related issue|
|4.3|
||

Description:
Fix for preserver cookie on test suite execution

### CODE REVIEW

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js [hunk 1]
```diff
@@ -63,18 +63,46 @@ export const clearSession = () => {
...
+            cy.setCookie(element.name, element.value);
+          })
+        })
+      }
+    });
+  } catch (e) {
+  }
 }
 
+// export const checkCookie = () => {
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/utils/driver.js:86]: do we need these commented lines? or are old deprecated methods?
if we don't use them please delete them.

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js [hunk 2]
```diff
@@ -83,18 +111,60 @@ export const updateCookies = () => {
...
         })
         cy.writeFile(filename,  JSON.stringify(newCookie))
-
       })
     }
+   }
   });
 }
 
+// export const updateCookies2 = () => {
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/utils/driver.js:125]: same here, commented functions should not be in the production code.

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js [hunk 3]
```diff
@@ -83,18 +111,60 @@ export const updateCookies = () => {
...
+
+//       })
+//     }
+//   });
+// }
+
+export const preserveCookie = () => {
+  let str = [];
+ return cy.getCookies().then((cook) => {
+      // cy.log('after - get cookie',cook);
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/utils/driver.js:151]: comment

**Review by MauGaP — CHANGES_REQUESTED** [review]

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js (hunk 2 above)
Comment (Mayons) [inline tests/frontend/test_kibana_app/cypress/integration/utils/driver.js:125]: Also here, there are necessary those comment lines?

File: tests/frontend/test_kibana_app/cypress/support/index.js [hunk 4]
```diff
@@ -42,38 +49,53 @@ before(() => {
...
     cy.setSessionStorage('healthCheck', 'executed');
-    updateExpiryValueCookies()
-    setCookies(cookieMock);
+    updateExpiryValueCookies();
+    preserveCookie()
+    updateCookies();
 })
 
 afterEach(() => {
     //Code to Handle the Sesssion cookie in cypress.
```
[the remark this answers is not in the document]
Comment (mauceballos) [reply tests/frontend/test_kibana_app/cypress/support/index.js]: Done

File: tests/frontend/test_kibana_app/cypress/support/index.js [hunk 5]
```diff
@@ -42,38 +49,53 @@ before(() => {
 
     login ? login() : cy.log(`Error! loginMethod: "${loginMethod}" is not recognized`);
 
-    cy.wait(1000);
+    cy.wait(15000);
 
     validateURLIncludes(OVERVIEW_URL);
 
+    // cy.getCookies().then((cook) => {
```
[the remark this answers is not in the document]
Comment (mauceballos) [reply tests/frontend/test_kibana_app/cypress/support/index.js]: done

File: tests/frontend/test_kibana_app/cypress/support/index.js [hunk 6]
```diff
@@ -15,9 +15,16 @@
 
 // Import commands.js using ES2015 syntax:
 
-import { navigate, validateURLIncludes, setCookies } from '../integration/utils/driver';
 import { LOGIN_TYPE, OVERVIEW_URL } from '../integration/utils/login-constants';
-import { updateCookies, clearSession, updateExpiryValueCookies } from '../integration/utils/driver';
+import {
```
[the remark this answers is not in the document]
Comment (mauceballos) [reply tests/frontend/test_kibana_app/cypress/support/index.js]: done

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js (hunk 3 above)
Comment (mauceballos) [reply tests/frontend/test_kibana_app/cypress/integration/utils/driver.js]: done

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js (hunk 2 above)
Comment (mauceballos) [reply tests/frontend/test_kibana_app/cypress/integration/utils/driver.js]: done

File: tests/frontend/test_kibana_app/cypress/integration/utils/driver.js (hunk 1 above)
Comment (mauceballos) [reply tests/frontend/test_kibana_app/cypress/integration/utils/driver.js]: done

**Review by Mayons — APPROVED** [review]

**Review by Mayons — APPROVED** [review]

**Review by MauGaP — APPROVED** [review]

**Comment by MauGaP** [comment]
LGTM

**Pull request merged**